dll/cpl/sysdm/general.c

```diff
@@ -598,73 +606,75 @@ static VOID GetSystemVersion(HWND hwnd)
     HeapFree(GetProcessHeap(), 0, pwszStr);
 }
 
-ULONGLONG GetSecondsQPC(VOID)
+static ULONGLONG GetTickCountQPC(VOID)
 {
     LARGE_INTEGER Counter, Frequency;
 
     QueryPerformanceCounter(&Counter);
     QueryPerformanceFrequency(&Frequency);
 
-    return Counter.QuadPart / Frequency.QuadPart;
+    return (Counter.QuadPart * 1000) / Frequency.QuadPart;
 }
 
-ULONGLONG GetSeconds(VOID)
+static VOID GetSystemUptime(HWND hwndDlg)
 {
-    ULONGLONG (WINAPI * pGetTickCount64)(VOID);
-    ULONGLONG Ticks64;
-    HMODULE hModule = GetModuleHandleW(L"kernel32.dll");
+    HWND hUptimeLabel;
+    WCHAR szBuf[64];
+    ULONG cSeconds;
 
-    pGetTickCount64 = (PVOID)GetProcAddress(hModule, "GetTickCount64");
     if (pGetTickCount64)
     {
-        return pGetTickCount64() / 1000;
+        RosUptime = pGetTickCount64();
     }
-
-    hModule = LoadLibraryW(L"kernel32_vista.dll");
-
-    if (!hModule)
+    else
     {
-        return GetSecondsQPC();
+        RosUptime = GetTickCountQPC();
     }
 
-    pGetTickCount64 = (PVOID)GetProcAddress(hModule, "GetTickCount64");
-
-    if (pGetTickCount64)
-    {
-        Ticks64 = pGetTickCount64() / 1000;
-    }
-    else
+    hUptimeLabel = GetDlgItem(hwndDlg, IDC_UPTIME);
+    if (!hUptimeLabel)
     {
-        Ticks64 = GetSecondsQPC();
+        return;
     }
+    
+    cSeconds = RosUptime / 1000;
+    StringCchPrintfW(szBuf, _countof(szBuf), szUptimeFormat,
+                     cSeconds / (60*60*24),
+                     (cSeconds / (60*60)) % 24,
+                     (cSeconds / 60) % 60,
+                     cSeconds % 60);
 
-    FreeLibrary(hModule);
-    return Ticks64;
+    SetWindowTextW(hUptimeLabel, szBuf);
 }
 
-VOID GetSystemUptime(HWND hwnd)
+static VOID InitSystemUptime(HWND hwndDlg)
 {
-    HWND hRosUptime;
-    WCHAR szBuf[64], szStr[64];
-    ULONG cSeconds;
-
-    hRosUptime = GetDlgItem(hwnd, IDC_UPTIME);
-    if (!hRosUptime)
+    /* Load time format string */
+    if (LoadStringW(hApplet, IDS_UPTIME_FORMAT, szUptimeFormat, _countof(szUptimeFormat)) == 0)
     {
         return;
     }
-    if (!LoadStringW(hApplet, IDS_UPTIME_FORMAT, szStr, _countof(szStr)))
+
+    /* Load required DLLs */
+    hKernel32 = GetModuleHandleW(L"kernel32.dll");
```

Comment:
Is there a reason to load kernel32.dll explicitly? It's already dynamically linked to it: https://github.com/reactos/reactos/blob/384a711e77375cd88e26801effde98df0d6d6fc3/dll/cpl/sysdm/CMakeLists.txt#L31

Reply:
Oops, nevermind, I missed it's just GetModuleHandleW. :sweat_smile: